### ISSUE TITLE
Gain and frames

### ISSUE THREAD

**Issue by twostage** [4br3mm0rd/mpyg321#10]
Hi,

Not really an issue but being able to set the volume (gain) and being able to play only the first N frames would be useful additions.

Thanks,

**Comment by TheTechRobo** [comment]
Yeah! Similarily, it would be really nice if we could start by playing the whole audio, then when we loop it, instead of starting over, we start at X frame of the audio.

**Comment by TheTechRobo** [comment]
Possibly related: `jump()` function

**Comment by twostage** [comment]
Made a local copy and added -

def gain(self, vol):
        """sets the volume"""
        self.player.sendline("GAIN " + str(vol))

I need to learn how github works don't I ?

**Comment by TheTechRobo** [comment]
You could FORK the repository (top-left), change it inside your fork, and then submit a Pull Request from your fork to this repository.

**Comment by 4br3mm0rd** [comment]
Hey! Thank you for your contribution, that would be a great idea!

Feel free to create a pull request :-)

**Comment by twostage** [comment]
> You could FORK the repository (top-left), change it inside your fork, and then submit a Pull Request from your fork to this repository.

Yep, I think I understand that. I'll give it a go. (Been programming in Python for years and never put anything on github)

**Comment by 4br3mm0rd** [comment]
If you want to add an ability to play only the first N frames, you can add a value in mpg_outs for "@F", and a listener frame_played (or something like that). The number following "@F" is the number of the frame. You can compare it to the number of frames you want to play (that you would save in a property of the MPyg321Player instance)

**Comment by 4br3mm0rd** [comment]
Hey everyone!
Anyone working/blocked on this issue? @twostage

**Comment by TheTechRobo** [comment]
> If you want to add an ability to play only the first N frames, you can add a value in mpg_outs for "@f", and a listener frame_played (or something like that). The number following "@f" is the number of the frame. You can compare it to the number of frames you want to play (that you would save in a property of the MPyg321Player instance)

I didn't follow that at all :sweat_smile: so I can't help :P

**Comment by 4br3mm0rd** [comment]
See #14 for GAIN :-) (actually VOLUME, and I think this is because of #15 - we will probably handle this/make a design choice in the coming days about this )

**Comment by 4br3mm0rd** [comment]
Volume is implemented in #18 for mpg123 and #19 for mpg321. Available in version 1.4.1

I'll try to implement the "jump to frame" feature in a near future :) 

PS: @stefets now I know why I wanted to keep the frame output :p ...

**Comment by stefets** [comment]
> PS: @stefets now I know why I wanted to keep the frame output :p ...

Yes! :)
It wasn't long before a use case appeared!

**Comment by hakmad** [comment]
#1 implements a function to jump either to a specific time, a specific frame, or an offset from the current position. For jumping to a specific frame you might do:

`player.jump(200)`

If you wanted to jump ahead 20 MPEG frames, you would do:

`player.jump("+20")`

You can take a look at the examples as well as running `mpg123 -R` and then typing `HELP` to get more information. :D
